testutil/pty.go

```diff
@@ -0,0 +1,64 @@
+package testutil
+
+import (
+	"context"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/hinshun/vt10x"
+)
+
+// ReadUntilString emulates a terminal and reads one byte at a time until we
+// either see the string we want, or the context expires.
+func ReadUntilString(ctx context.Context, t *testing.T, want string, r io.Reader) error {
+	return ReadUntil(ctx, t, r, func(line string) bool {
+		return strings.TrimSpace(line) == want
+	})
+}
+
+// ReadUntil emulates a terminal and reads one byte at a time until the matcher
+// returns true or the context expires.  If the matcher is nil, read until EOF.
+func ReadUntil(ctx context.Context, t *testing.T, r io.Reader, matcher func(line string) bool) error {
+	// output can contain virtual terminal sequences, so we need to parse these
+	// to correctly interpret getting what we want.
+	term := vt10x.New(vt10x.WithSize(80, 80))
```

Comment:
I'm nervous about hardcoding these values when this function is in `testutil`.  If someone creates a PTY with a different size, then tries to use this, they could get unexpected results.  Adding them as required parameters will hurt readability, and it seems superfluous to make them options that we don't actually use yet.

At the very least, the comment on the function should warn users that their PTY must be sized to 80x80.